lib/Runtime/Library/JavascriptLibrary.h

```diff
@@ -55,6 +55,13 @@ namespace Js
         Field(int) validPropStrings;
     };
 
+    struct ObjectCreateTypeCache
+    {
+        static const uint MaxCachedTypes = 32;
+        Field(RecyclerWeakReference<RecyclableObject>*) prototype;
+        Field(RecyclerWeakReference<DynamicType>*) type;
+    };
+
```

Comment:
Would it be possible to wrap the array of cache entries into a full struct/class?  We'd get better encapsulation, and we could also move the cache lookup logic (lines 7296-7299 of JavascriptLibrary.cpp) into the cache implementation, again for better encapsulation.

Reply:
done